src/ameba/ast/branch.cr

```diff
@@ -172,6 +176,16 @@ module Ameba::AST
       def end_visit(node : Crystal::MacroFor)
         on_branchable_end node
       end
+
+      def visit(node : Crystal::Call)
+        if loop?(node) && (block = node.block)
+          on_branchable_start node, block.body
+        end
+      end
+
+      def end_visit(node : Crystal::Call)
+        on_branchable_end node if loop?(node)
```

Comment:
i think this `if` condition should match the one in visit method because it can mess up the scope